runtime-light/coroutine/shared-task.h

```diff
@@ -57,20 +57,20 @@ struct promise_base_t {
         return waiter->m_continuation;
       }
 
-      constexpr void await_resume() const noexcept {}
+      constexpr auto await_resume() const noexcept -> void {}
     };
-    return final_suspend_t{};
+    return awaiter{};
   }
 
-  constexpr void unhandled_exception() noexcept {
+  auto unhandled_exception() const noexcept -> void {
     php_critical_error("internal unhandled exception");
   }
 
-  constexpr bool ready() const noexcept {
+  auto done() const noexcept -> bool {
     return m_waiters == this;
   }
 
-  constexpr void add_ref() noexcept {
+  auto add_ref() noexcept -> void {
```

Comment:
Why do you use such a style `auto foo() -> Bar {...}`?
As I see, it isn't commonly used in repo

Reply:
I decided to use it here to increase readability. It's not so important to first see the return type of coroutine traits functions. In opposite it's crucial to clearly see the structure behind these functions, e.g. clearly separate awaiters and promises, initial and final suspenders etc